src/app/contact/contact.component.html

```diff
@@ -0,0 +1,174 @@
+<div class="container">
+  <div style="margin-top: 50px; margin-left: 20%; margin-right: 20%;">
+    <div class="panel panel-default">
+      <div class="panel-heading">
+        <h3 style="text-align: center;">Contact Form</h3>
+      </div>
+      <div class="panel-body">
+        <div class="row">
+          <div class="col-md-8 offset-md-2">
+            <form novalidate (ngSubmit)="Save()" [formGroup]="contactForm">
+              <div class="form-group">
+                <label for="firstNameId">Title</label>
+                <div>
+                  <select class="form-control" id="title" formControlName="title">
+                    <option *ngFor="let title of titles" [value]="title">{{
+                      title
+                    }}</option>
+                  </select>
+                </div>
+              </div>
+              <div class="form-group">
+                <label for="firstNameId">First Name</label>
+                <div>
+                  <input class="form-control" id="firstNameId" type="text" placeholder="First Name (required)"
+                    formControlName="firstName" [ngClass]="{
+                      'is-invalid':
+                        (contactForm.get('firstName').touched ||
+                          contactForm.get('firstName').dirty) &&
+                        !contactForm.get('firstName').valid
+                    }" />
+                  <span class="invalid-feedback">
+                    <span *ngIf="contactForm.get('firstName').errors?.required">
+                      First name is required.
+                    </span>
+                    <span *ngIf="contactForm.get('firstName').errors?.pattern">
+                       first name must be longer than 2 characters.
+                    </span>
+                    <span *ngIf="
+                        contactForm.get('firstName').errors?.textvalidation
+                      ">
+                      No numbers allowed,letters only
+                    </span>
+                  </span>
+                </div>
+              </div>
+
+              <div class="form-group">
+                <label for="lastNameId">Last Name</label>
+                <div>
+                  <input class="form-control" id="lastNameId" type="text" placeholder="Last Name (required)"
+                    formControlName="lastName" [ngClass]="{
+                      'is-invalid':
+                        (contactForm.get('lastName').touched ||
```

Comment:
Please move the calls to `get` inside the component.

Reply:
@strizzwald fixed